roles/translator/src/status.rs

```diff
@@ -1,12 +1,44 @@
 use crate::error::Error;
 
 #[derive(Debug)]
-pub enum State {
-    Shutdown(Error),
+pub enum State<'a> {
+    Shutdown(Error<'a>),
     Healthy(String),
 }
 
 #[derive(Debug)]
-pub struct Status {
-    pub state: State,
+pub struct Status<'a> {
+    pub state: State<'a>,
+}
+
+/// # Description
+/// This macro handles errors inserting error handling logic for a given `Result<T, crate::error::Error<'a>>`
+/// it is used by passing in a `Sender<crate::status::Status>` as the first parameter
+/// and a `Result<T, crate::error::Error<'a>>` as the second parameter
+/// # Example
+/// ```
+/// let (tx_status: Sender<Status>, rx_status) = async_channel::unbounded();
+/// let variable = handle_result!(
+///     tx_status,
+///     type_.try_into()
+/// );
+/// ```
+#[macro_export]
+macro_rules! handle_result {
```

Comment:
Can we use timelines to ensure that this is ONLY called from within an async task?